From cb60f55a8eb26753e221f3ffe9cbf667e4a5074f Mon Sep 17 00:00:00 2001
From: Yasufumi Ogawa <yasufum.o@gmail.com>
Date: Sun, 16 Feb 2020 14:34:02 +0900
Subject: [PATCH 11/29] tools/sppc: check DPDK ver in load-balancer

This update is to add checking DPDK version in load-balancer app
launcher because this example app was removed from v19.08-rc1. If DPDK
version on container is larger than v19.08-rc1, launcher stops
launching.

Signed-off-by: Yasufumi Ogawa <yasufum.o@gmail.com>
---
 tools/sppc/app/load-balancer.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/tools/sppc/app/load-balancer.py b/tools/sppc/app/load-balancer.py
index d86a04b..ec504e7 100755
--- a/tools/sppc/app/load-balancer.py
+++ b/tools/sppc/app/load-balancer.py
@@ -68,6 +68,16 @@ def main():
             common.IMG_BASE_NAMES['dpdk'],
             args.dist_name, args.dist_ver)
 
+    c_dpdk_ver = app_helper.get_dpdk_ver_in_container(
+            env.RTE_SDK, container_image)
+    expected = '19.08-rc1'
+    if app_helper.compare_version(expected, c_dpdk_ver) > 0:
+        print("Load-balancer example was removed after DPDK 'v{}'.".
+              format(expected))
+        print("You cannot run it because DPDK in the container is 'v{}'.".
+              format(c_dpdk_ver))
+        exit()
+
     # Setup devices with given device UIDs.
     dev_uids = None
     sock_files = None
-- 
2.17.1

